lib/search.js

```diff
@@ -53,8 +54,13 @@ function doSearch(query) {
         query.apps = query.apps.concat(apps);
         return query;
     }
-
-    return request(buildUrl(query))
+    var delay = h.calcDelay(optsGlobal.throttle, request._last);
+    return Promise
```

Comment:
i don't see promise defined twice, can you point it please?

if we'll handle delay _after_ request - this wont be fair delay (actual throttle minus network delay)
now we are making delay and _immediately_ call google api
my opinion: delaying little more is much better for business logic than little less